### DISCUSSION TITLE
Cascade Delete from UI

### DISCUSSION THREAD

**Discussion by odama626** [supabase/supabase#6979] [Feature Requests] [answered]
Allow setting cascade delete on foreign keys

**Comment by MildTomato** [comment]
Will move this to 'ideas' in discussion

**Reply by farhanhaider1** [reply]
> Allow setting cascade delete on foreign keys

+1000000 😆

**Reply by MildTomato** [reply]
I can confirm that this is now live and supported in Supabase Studio

**Comment by anngbaum** [comment]
just wanted to add a +1 and mention that this comes up frequently in discord ([here](https://discord.com/channels/839993398554656828/843999948717555735/966349081515544677) and [here](https://discord.com/channels/839993398554656828/843999948717555735/959744851753173114) to name a few, but searching `on delete cascade` shows a number of examples).

**Comment by vimtor** [comment]
I don't know why, but I expected Supabase to delete on cascade by default since it is pretty common in SQL databases. I think it could fit in the cog icon popover or in the slide over when setting up a foreign key relationship 😄

**Comment by rbkayz** [comment]
Yes please. This would be great. Having to delete constraint, and add it is a bit cumbersome

**Comment by codeibrahima** [comment]
+1 to this

**Comment by hnrq** [comment]
+1, definitely need that

**Comment by bennik88** [comment]
Would be an important addition!

**Comment by harryc9** [comment]
This feature would be amazing!! (esp for me who doesn't want to deal with the SQL side of things 🤭)

**Comment by ahmedDev20** [comment]
+1

**Comment by jhb-dev** [comment]
+1

**Comment by nenek** [comment]
+1

**Comment by Isaac-Harper** [comment]
+1

**Comment by jjrise** [comment]
+1

**Comment by alexrabin** [comment]
+100000000000000

**Comment by VenierGiacomo** [comment]
+1

**Comment by JoleMile** [comment]
+1

**Comment by hmalkawi1** [comment]
+12223213232

**Comment by cdedreuille** [comment]
Can't agree more. I would love to see that happen. Would make the deleting process so much easier.

**Comment by RyanTippsTX** [comment]
Struggling with the same issue right now

**Comment by RyanTippsTX** [comment]
or at least allow an option from service role:
auth.admin.deleteUser(host_id, {
    cascade: true,
})

**Comment by hkleine** [comment]
+1

**Comment by Ruud14** [comment]
+1

**Comment by RyanTippsTX** [comment]
In Postgres, the ON DELETE CASCADE feature is specified for any foreign key
relation in the table schema itself (not as a modifier to a DELETE
statement), so the supabase app’s table editor UI might be a more
reasonable place to extend this feature to users.

**Comment by othema** [comment]
+1

**Comment by bryanherediax** [comment]
+1

**Comment by N00nDay** [comment]
Does anyone have a workaround for this issue? I can't even get `ON CASCADE` working through SQL. I found the snippet below:
```
ALTER TABLE public.join_tags
DROP CONSTRAINT join_tags__customer_id_fkey,
ADD CONSTRAINT join_tags__customer_id_fkey
    FOREIGN KEY (id)
    REFERENCES public.customers(id)
    ON DELETE CASCADE;
```
but this returns an error that violates the foreign key constraint.

**Reply by N00nDay** [reply]
Finally figured it out. I had to do the following to get it to work:

1. Remove the foreign key from the `_customer_id` column
2. Run the function below:
```
ALTER table public.join_tags
ADD foreign KEY (_customer_id) references customers(id) on delete cascade;
```

The only way I could get it to work.

**Reply by SlingingCode** [reply]
Thanks

**Comment by bryanherediax** [comment]
+10000 for this.

**Comment by alexgshaw** [comment]
+1

**Comment by SlingingCode** [comment]
+1

**Comment by alexrabin** [comment]
Has someone written a script to get the total amount of all the "+1"s & "+10000"s yet?

If not I'll get on that asap.

**Comment by alexrabin** [comment]
> Has someone written a script to get the total amount of all the "+1"s & "+10000"s yet?
> 
> If not I'll get on that asap.

Alrighty, it's up. https://supabase-discussion-tracker.netlify.app

**Comment by teddarcuri** [comment]
+42069

**Comment by alvaroartano** [comment]
+2506498762

**Comment by lambou** [comment]
cascade, restrict, and others

**Comment by alvaroartano** [comment]
+9999*99999999**99999999999999

**Reply by alvaroartano** [reply]
![image](https://user-images.githubusercontent.com/28715114/207262723-6d9cd30f-23b2-4b4b-a4c5-27990bd43654.png)
😎

**Reply by alexrabin** [reply]
Thanks for breaking it 😂

**Comment by Edo-San** [comment]
+1!

**Comment by dave-hay** [comment]
+1

**Comment by dougwithseismic** [comment]
+2023

**Comment by edymusajev** [comment]
+1

How is this not a feature? 😅

**Comment by EskelCz** [comment]
Please apply this to deleting users from auth

**Comment by sasweb** [comment]
+42

**Comment by rbkayz** [comment]
A few recommendations:

1. Definitely install [pgAdmin](https://www.pgadmin.org/) - cascade is just one part of what you'll want to maintain. Ideally, other constraints, indexes, etc. are better handled there (till supabase adds options on the console)

2. If you want to set cascade delete to a FK, run this script in your console SQL tab:

```
ALTER TABLE <table_name>DROP CONSTRAINT IF EXISTS <constraint_name> 

ALTER TABLE <table_name> ADD CONSTRAINT <constraint_name> FOREIGN KEY (<col_name>) REFERENCES other_tbl_name (<other_col_name>) ON DELETE CASCADE;
```

**Comment by Shenrak** [comment]
+1
Please...

**Comment by rubywai** [comment]
+1

**Comment by sannajammeh** [comment]
+1

**Comment by eialanjones** [comment]
+1 and counting...

**Comment by RizanPSTU** [comment]
+1 please

**Comment by MattConflitti** [comment]
+1

**Comment by StefanVDWeide** [comment]
This has been open for almost a year now, with mostly "+1" answers. Can anyone maybe shine some light if this feature is on any of the roadmaps? And if so, what timeline are we looking at? 

Having this built in is infinitely better than rolling my own solution!

**Reply by alvaroartano** [reply]
It is actually on their roadmap, but they are not throwing any estimates, so we are just waiting to it... I think it is quite an easy thing to solve, so open source contributors could solve this easily if they knwo hwo to web dashbaord works.

But at the moment, we will have to wait.

**Reply by sasweb** [reply]
Technically it's an easy task but it's definitely not that easy to solve. 

You need to decide about certain UX flows and make the cascade also editable which requires changes in several views. It's always hard for an "outsider" to make these kind of updates without alignment with the internal supabase UX/UI team. 

Otherwise I would have already tackled it :/

**Comment by benrugg** [comment]
Just gonna have to keep adding +1's to this...

**Comment by benrugg** [comment]
🔥 For future visitors with this problem, here's a great resource that will give you the exact sql statement you need for your specific table/column names: https://index.garden/a/supabase-cascade/#supabase-delete-cascade

**Comment by onurusluca** [comment]
Please!!! +!

**Comment by arshamg** [comment]
This is a sorely needed feature. Even better would be one-click ability to turn this on for all tables in `public` schema for `user_id`. I think that is a very common use case (deleting a user and associated records).

**Comment by gutsyphilip** [comment]
+1

**Comment by onurusluca** [comment]
Good news! It looks like it is being implemented.
https://github.com/supabase/supabase/pull/12846

**Comment by justinwride** [comment]
This is now live 🚀 
I can confirm that I am able to edit any foreign key relation in the UI to select the action for "on delete"
https://github.com/supabase/supabase/pull/12846

**Reply by raphaelbarbosaqwerty** [reply]
Finally :rocket:

**Comment by harryc9** [comment]
We did it everyone!! 🙌

**Comment by MildTomato** [comment, marked as answer]
I can confirm that this is now live and supported in Supabase Studio
